src/widgetastic_patternfly4/modal.py

```diff
@@ -0,0 +1,60 @@
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import Text, Widget
+from widgetastic.xpath import quote
+from widgetastic_patternfly4 import Button
+
+
+class ModalItemNotFound(Exception):
+    pass
+
+
+class Modal(Widget):
```

Comment:
Because you include other widgets it's better to inherit `View`. Another option would be just define locators:
```python
TITLE = ".//h1[contains(@class, 'pf-c-title')]"
CLOSE = ".//button[@aria-label='Close']"
...
@property
def title(self):
    """Get title of modal window."""
    return self.browser.text(self.browser.element(self.TITLE))
...
def close(self):
    """Close modal window."""
    self.browser.click(self.browser.element(self.CLOSE))
```

Reply:
thanks, I didn't know what to inherit, so I just used what I saw in other components. Time to look at how widgetastic works